agp-patch/src/main/kotlin/app/cash/better/dynamic/features/AgpPatchPlugin.kt

```diff
@@ -16,15 +16,27 @@
 package app.cash.better.dynamic.features
 
 import com.android.Version
+import com.android.build.gradle.internal.crash.afterEvaluate
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 
 class AgpPatchPlugin : Plugin<Project> {
   override fun apply(target: Project) {
+    val extension = target.extensions.create("betterDynamicFeaturesPatch", BetterDynamicFeaturesPatchExtension::class.java)
+
     target.plugins.withId("com.android.application") {
-      // Do a strict version check to ensure that our monkey-patching will work correctly.
-      check(Version.ANDROID_GRADLE_PLUGIN_VERSION == TARGET_AGP_VERSION) {
-        "This version of the Android Gradle Plugin (${Version.ANDROID_GRADLE_PLUGIN_VERSION}) is not supported by the better-dynamic-features plugin. Only version $TARGET_AGP_VERSION is supported."
+      target.afterEvaluate {
```

Comment:
any chance this might not be needed, given that this callback should be called only when modules apply the application plugin?  i'm always reluctant to add more `afterEvaluate` blocks 😓

Reply:
or maybe push it out of here into its own global check, if it'd be a good idea to ensure an Android plugin is always applied for a module applying this plugin:

https://github.com/cashapp/paparazzi/blob/6eb59e36b756476fe212e38b9dd3df507f56b5f4/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt#L56

Reply:
In our use case, the agp-patch plugin is only applied to our main application module so that this version check runs.
The actual work of applying the "patch" is done by just including the plugin on the build classpath.

technically this block will also only run if the `com.android.application` plugin is applied in the first place! (there's a `plugins.withId()` block)